lock-keeper-client/src/api/generate_secret.rs

```diff
@@ -28,40 +28,24 @@ impl LockKeeperClient {
         let storage_key = self.retrieve_storage_key(request_id).await?;
 
         // Generate step: get new KeyId from server
-        let key_id = get_key_id(&mut channel, self.user_id()).await?;
+        let key_id_result: server::Generate = channel.receive().await?;
```

Comment:
```suggestion
        let key_id = channel.receive::<server::Generate>().await?.key_id;
```

If you want to store the key_id directly here, you could do it like this.